Comments from #287 addressed

tmcguire-rti opened the pull request:
I addressed all the comments. There are two issues.

1. I am not sure why git thinks RSAdapterPlugin files are new.

2. There is an issue seting the CMAKE_MODULE_PATH with "${CMAKE_CURRENT_SOURCE_DIR}/../../../../../resources/cmake". For the rs_main folder I want to build a Debug version. Unfortunately the generated VS project links to the release Connext Libraries which causes a runtime exception. If I set it to $NDDSHOME/resource/cmake the VS project correctly links the debug Connext DDS lLibraries.

<!--
:warning: Please, try to follow the template.
:warning: Your pull request title should be short, detailed and understandable for all.
:warning: If your pull request fixes an open issue, please link to the issue.

:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.
-->

### Summary



### Details and comments

iblancasa commented:
@tmcguire-rti, please, could you provide me a log output for the build and execution when the "runtime exception" appears?
We need to debug it.

tmcguire-rti commented:
Hello Israel.



In this example I am building an Debug executable that instantiates an
instance of a routing service. Unfortunately the build.txt does not show
the Debug configuration is linking to the release libraries. However you
can see from the attached the Debug configuration is referencing the
release Connext libraries. When I run the Debug example with the Release
libraries, I get the attached errors. If I change the Debug configuration
to link the Debug libraries and rebuild, it works as expected.







Regards,

Tim



*From:* Israel Blancas <notifications@github.com>
*Sent:* Wednesday, March 11, 2020 12:27 PM
*To:* rticommunity/rticonnextdds-examples <
rticonnextdds-examples@noreply.github.com>
*Cc:* tmcguire-rti <tmcguire@rti.com>; Mention <mention@noreply.github.com>
*Subject:* Re: [rticommunity/rticonnextdds-examples] Comments from #287
addressed (#288)



@tmcguire-rti <https://github.com/tmcguire-rti>, please, could you provide
me a log output for the build and execution when the "runtime exception"
appears?
We need to debug it.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/rticommunity/rticonnextdds-examples/pull/288#issuecomment-597733430>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AKOCBM6TJHNJHXSF4L2BLL3RG63WFANCNFSM4LFHY6LQ>
.

C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build>cmake -DBUILD_SHARED_LIBS=ON -DCONNEXTDDS_ARCH=x64Win64VS2017 -DCMAKE_BUILD_TYPE=Debug -G "Visual Studio 16 2019" -A x64 ..

-- The C compiler identification is MSVC 19.24.28316.0
-- The CXX compiler identification is MSVC 19.24.28316.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- RTI Connext DDS installation directory: C:/opt/rti_connext_dds-6.0.1
-- RTI Connext DDS architecture: x64Win64VS2017
-- Found RTIConnextDDS: C:/opt/rti_connext_dds-6.0.1 (found suitable version "6.0.1", minimum required is "6.0.1") found components:  core routing_service 
-- Found ODBC: odbc32.lib  
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/build

C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build>cmake --build . --config Debug

Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  CMake does not need to re-run because C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/build/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/CMakeLists.txt
  CMake does not need to re-run because C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/build/CMakeFiles/generate.stamp is up-to-date.
  rsmain.cxx
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/DiscoveryStreamReaderForwarder.hpp(40,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/transf/detail/TransformationForwarder.hpp(155,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/processor/Input.hpp(236,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/processor/Output.hpp(156,1): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(446,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(239): message : see reference to function template instantiation 'void rti::routing::adapter::detail::StreamReaderForwarder::proxy_read<false,false>(void *,RTI_RoutingServiceSample **,RTI_RoutingServiceSampleInfo **,int *,const RTI_RoutingServiceSelectorState *,RTI_RoutingServiceEnvironment *)' being compiled [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
  RsAdapterPlugin.cxx
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/DiscoveryStreamReaderForwarder.hpp(40,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(446,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(239): message : see reference to function template instantiation 'void rti::routing::adapter::detail::StreamReaderForwarder::proxy_read<false,false>(void *,RTI_RoutingServiceSample **,RTI_RoutingServiceSampleInfo **,int *,const RTI_RoutingServiceSelectorState *,RTI_RoutingServiceEnvironment *)' being compiled [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
  RsConnection.cxx
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/DiscoveryStreamReaderForwarder.hpp(40,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(446,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(239): message : see reference to function template instantiation 'void rti::routing::adapter::detail::StreamReaderForwarder::proxy_read<false,false>(void *,RTI_RoutingServiceSample **,RTI_RoutingServiceSampleInfo **,int *,const RTI_RoutingServiceSelectorState *,RTI_RoutingServiceEnvironment *)' being compiled [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
  RsStreamWriter.cxx
  RsStreamReader.cxx
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/DiscoveryStreamReaderForwarder.hpp(40,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(446,1): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
C:\opt\rti_connext_dds-6.0.1\include\rti/routing/adapter/detail/StreamReaderForwarder.hpp(239): message : see reference to function template instantiation 'void rti::routing::adapter::detail::StreamReaderForwarder::proxy_read<false,false>(void *,RTI_RoutingServiceSample **,RTI_RoutingServiceSampleInfo **,int *,const RTI_RoutingServiceSelectorState *,RTI_RoutingServiceEnvironment *)' being compiled [C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.vcxproj]
  RsInputDiscoveryStreamReader.cxx
  ConnectorBase.cxx
  ConnectorMsqlDb.cxx
  Generating Code...
     Creating library C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/build/rsmain.lib and object C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/build/rsmain.exp
  rsmain.vcxproj -> C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build\rsmain.exe
  Building Custom Rule C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/CMakeLists.txt
  CMake does not need to re-run because C:/Users/tmcguire/Documents/GitHub/rticonnextdds-examples/examples/routing_service/routing_service_msql_adapter/cpp/rs_main/build/CMakeFiles/generate.stamp is up-to-date.

C:\Users\tmcguire\Documents\GitHub\rticonnextdds-examples\examples\routing_service\routing_service_msql_adapter\cpp\rs_main\build>rsmain.exe ../MainRsMsqlShapesAdapter_w_types.xml RS_DDS_To_MSQL

|| RsAdapterPlugin || T ||> CALL create_connection
|| RsConnection || T ||> CALL input_stream_discovery_reader
|| RsConnection || T ||> CALL create_stream_writer
|| ConnectorMsqlDb || T ||> CALL set_connection_info
|| ConnectorMsqlDb || T ||> CALL connect_db
|| ConnectorMsqlDb || I ||> DRIVER={SQL Server}; SERVER=RTI-10591\SQLEXPRESS; DATABASE=Shapes; UID=rti_shapes_rs; PWD=abc123;
|| ConnectorMsqlDb || I ||> SQL connection to database successful
|| RsConnection || T ||> CALL create_stream_writer
|| ConnectorMsqlDb || T ||> CALL set_connection_info
|| ConnectorMsqlDb || T ||> CALL connect_db
|| ConnectorMsqlDb || I ||> DRIVER={SQL Server}; SERVER=RTI-10591\SQLEXPRESS; DATABASE=Shapes; UID=rti_shapes_rs; PWD=abc123;
|| ConnectorMsqlDb || I ||> SQL connection to database successful
|| RsConnection || T ||> CALL create_stream_writer
|| ConnectorMsqlDb || T ||> CALL set_connection_info
|| ConnectorMsqlDb || T ||> CALL connect_db
|| ConnectorMsqlDb || I ||> DRIVER={SQL Server}; SERVER=RTI-10591\SQLEXPRESS; DATABASE=Shapes; UID=rti_shapes_rs; PWD=abc123;
|| ConnectorMsqlDb || I ||> SQL connection to database successful
|| RsInputDiscoveryStreamReader || T ||> CALL take
Enter Ctrl-C to quit
|| RsInputDiscoveryStreamReader || T ||> CALL return_loan
|| RsStreamWriter || T ||> CALL write
DL Error: : DDS_DynamicData2StructPlugin_getMemberInfo: Cannot find a member (name = "PAµT┬☻", id = 0) in type XMLTypes::ShapeTypeExtended
DL Error: : DDS_DynamicData2_get_long: Invalid member request.
DL Error: : [/routing_services/RS_DDS_To_MSQL/domain_routes/DDS_to_MSQL_Route/sessions/dds_to_msql_session/routes/Squares Route/outputs/MsqlSquares] ROUTERStreamWriter_writeI:Failed to get int32_t value: member doesn't exist; retcode=0, stream=Square
DL Error: : ROUTERRoutingProcessor_routeToMatchingOutputs:!write on output at index=0: dropped samples count=1
DL Error: : ROUTERRoutingProcessor_onRouteData:!route from input to matching outputs
DL Error: : ROUTERRoutingProcessor_onRouteEvent:!route data
DL Error: : ROUTERProcessor_onRouteEvent:processor 'rti.routingservice.RoutingProcessor' error (0): Failed to get int32_t value: member doesn't exist
DL Error: : [/routing_services/RS_DDS_To_MSQL/domain_routes/DDS_to_MSQL_Route/sessions/dds_to_msql_session/routes/Squares Route] ROUTERTopicRoute_processEvent:!process event=DATA_ON_INPUTS
DL Error: : ROUTERTopicRoute_onConditionTriggered:!process event
|| RsStreamWriter || T ||> CALL write
DL Error: : DDS_DynamicData2StructPlugin_getMemberInfo: Cannot find a member (name = " NµT┬☻", id = 0) in type XMLTypes::ShapeTypeExtended
DL Error: : DDS_DynamicData2_get_long: Invalid member request.
DL Error: : [/routing_services/RS_DDS_To_MSQL/domain_routes/DDS_to_MSQL_Route/sessions/dds_to_msql_session/routes/Squares Route/outputs/MsqlSquares] ROUTERStreamWriter_writeI:Failed to get int32_t value: member doesn't exist; retcode=0, stream=Square
DL Error: : ROUTERRoutingProcessor_routeToMatchingOutputs:!write on output at index=0: dropped samples count=1
DL Error: : ROUTERRoutingProcessor_onRouteData:!route from input to matching outputs
DL Error: : ROUTERRoutingProcessor_onRouteEvent:!route data
DL Error: : ROUTERProcessor_onRouteEvent:processor 'rti.routingservice.RoutingProcessor' error (0): Failed to get int32_t value: member doesn't exist
DL Error: : [/routing_services/RS_DDS_To_MSQL/domain_routes/DDS_to_MSQL_Route/sessions/dds_to_msql_session/routes/Squares Route] ROUTERTopicRoute_processEvent:!process event=DATA_ON_INPUTS
